sequencer/src/l1_client.rs

```diff
@@ -88,167 +85,57 @@ impl Committable for L1BlockInfo {
 }
 
 #[derive(Clone, Debug)]
-pub struct L1ClientOptions {
-    url: Url,
-    finalized_block_tag: BlockNumber,
+/// An Http Provider and configuration to interact with the L1.
+pub struct L1Client {
     retry_delay: Duration,
+    /// `Provider` from `ethers-provider`.
+    provider: Provider<Http>,
 }
 
-impl L1ClientOptions {
-    pub fn with_url(url: Url) -> Self {
+impl L1Client {
+    /// Instantiate an `L1Client` for a given `Url`.
+    pub fn new(url: Url) -> Self {
         Self {
-            url,
-            finalized_block_tag: BlockNumber::Finalized,
             retry_delay: Duration::from_secs(1),
+            provider: Provider::new(Http::new(url)),
         }
     }
-
-    pub fn with_latest_block_tag(mut self) -> Self {
-        // For testing with a pre-merge geth node that does not support the finalized block tag we
-        // allow setting the client to use the latest block instead.
-        self.finalized_block_tag = BlockNumber::Latest;
-        self
-    }
-
-    pub async fn start(self) -> Result<L1Client, ProviderError> {
-        let rpc = Provider::connect(self.url.clone()).await?;
-
-        // Fetch the initial L1 snapshot now, blocking, so that we will have a reasonable baseline
-        // at least until the background task starts updating the snapshot.
-        let snapshot = L1Snapshot {
-            head: rpc.get_block_number().await?.as_u64(),
-            finalized: get_finalized_block(&rpc, self.finalized_block_tag).await?,
-        };
-        tracing::info!("L1 client initialized with snapshot {snapshot:?}");
-
-        let client = L1Client {
-            latest_snapshot: Arc::new(RwLock::new(snapshot)),
-        };
-
-        // Spawn a background task to update the in-memory snapshot whenever a new L1 block is
-        // produced.
-        spawn(update_loop(
-            self.url,
-            self.finalized_block_tag,
-            self.retry_delay,
-            client.latest_snapshot.clone(),
-        ));
-
-        Ok(client)
-    }
-}
-
-#[derive(Clone, Debug)]
-pub struct L1Client {
-    latest_snapshot: Arc<RwLock<L1Snapshot>>,
-}
-
-impl L1Client {
+    /// Get a snapshot from the l1, or a fake snapshot if "testing"
```

Comment:
Is it still true that we provide a fake snapshot if the "testing" feature is enabled?

Reply:
It is not true. Fixed in d7963d1e42805a3f5257fba6515457409ccfa8d4